File: Tec/8vo semestre/AutomatasII/arbol_balanceado/arbol.rb
require './nodo_arbol'

class Arbol

    def initialize
        @arbol = nil
    end

    def agregar_nodo dato,nombre
        nuevo = NodoArbol.new(dato,nombre)

        if @raiz == nil
            @raiz = nuevo
            puts "Raiz --> [#{@raiz.dato}]"
        else
            auxiliar = @raiz
            padre = nil

            while true
                padre = auxiliar
                
                if dato < auxiliar.dato
                    auxiliar = auxiliar.hijo_izquierdo
                    if auxiliar == nil
                        padre.hijo_izquierdo = nuevo
                        puts "[#{padre.dato} --- izquierda --> [#{padre.hijo_izquierdo.dato}]"
                        return
                    end
                else
                    auxiliar = auxiliar.hijo_derecho
                    if auxiliar == nil
                        padre.hijo_derecho = nuevo
                        puts "[#{padre.dato} --- derecha --> [#{padre.hijo_derecho.dato}]"
                        return
                    end
                end
            end
        end
    end
end

File: Tec/8vo semestre/AutomatasII/ArbolBalanceado/src/principal/Arbol.java
package principal;

public class Arbol {

    NodoArbol raiz;

    public Arbol() {
        this.raiz = null;
    }

    public void agregarNodo(int dato, String nombre) {
        NodoArbol nuevo = new NodoArbol(dato, nombre);

        if (raiz == null) {
            raiz = nuevo;
            System.out.println("raiz --> ["+raiz.dato+"]");
        } else {
            NodoArbol auxiliar = raiz;
            NodoArbol padre;

            while (true) {
                padre = auxiliar;

                if (dato < auxiliar.dato) {
                    auxiliar = auxiliar.hijoIzquierdo;
                    if(auxiliar == null){
                        padre.hijoIzquierdo = nuevo;
                        System.out.println("["+padre.dato+"]---izquierda-->["+padre.hijoIzquierdo.dato+"]");
                        return;
                    }
                }else{
                    auxiliar = auxiliar.hijoDerecho;
                    if(auxiliar == null){
                        padre.hijoDerecho = nuevo;
                        System.out.println("["+padre.dato+"]---derecha-->["+padre.hijoDerecho.dato+"]");
                        return;
                    }
                }
            }
        }
    }
}
